js/views/search/Search.js

```diff
@@ -0,0 +1,248 @@
+
+export default class extends baseVw {
+  constructor(options = {}) {
+    super(options);
+    this.options = options;
+
+    const term = options.term;
+    const testForURL = /^((http|https|ob):\/\/)/;
+
+    if (term && testForURL.test(term)) {
+      // if a search URL was passed in, reconstruct the url and parse the data
+      const searchURL = new URL(`${term}${options.query ? `?${options.query}` : ''}`);
+      const params = searchURL.searchParams;
+      this.sProvider = `${searchURL.origin}${searchURL.pathname}`;
+      this.serverPage = params.get('p') || 0;
+      this.pageSize = params.get('ps') || 12;
+      this.term = params.get('q') || '';
+      this.callSearchProvider(searchURL);
+    } else {
+      this.sProvider = app.localSettings.get('searchProvider');
+      this.serverPage = options.serverPage || 0;
+      this.pageSize = options.pageSize || 12;
+      // if the term was not a url, process the term before calling the search provider
+      this.processTerm(term);
+    }
+
+    this.usingDefault = this.sProvider === app.localSettings.get('searchProvider');
+
+    // if not using a passed in URL, update the default provider if it changes
+    this.listenTo(app.localSettings, 'change:searchProvider', (_, provider) => {
+      if (this.usingDefault) {
+      this.sProvider = provider;
+      this.processTerm(this.term);
+    }
+  });
+  }
+
+  className() {
+    return 'search';
+  }
+
+  events() {
+    return {
+      'click .js-searchBtn': 'clickSearchBtn',
+      'change .js-sortBy': 'changeSortBy',
+      'change .js-filterWrapper select': 'changeFilter',
+      'change .js-filterWrapper input': 'changeFilter',
+      'keyup .js-searchInput': 'onKeyupSearchInput',
+    };
+  }
+
+  get sortByQuery() {
+    // return current sortBy state in the form of a query string
+    return this.$sortBy && this.$sortBy.length ? `&sortBy=${this.$sortBy.val()}` : '';
+  }
+
+  get filterQuery() {
+    // return all currently active filters in the form of a query string
+    return this.$filters && this.$filters.length ? `&${this.$filters.serialize()}` : '';
+  }
+
+  processTerm(term) {
+    this.term = term || '';
+    // if term has spaces, replace them with +
+    // if term is false, search for *
+    const query = `q=${term ? term.replace(/\s+/g, '+') : '*'}`;
+    const page = `&p=${this.serverPage}&ps=${this.pageSize}`;
+    const searchURL = `${this.sProvider}?${query}${this.sortByQuery}${this.filterQuery}${page}`;
+
+    this.callSearchProvider(searchURL);
+  }
+
+  callSearchProvider(searchURL) {
+    // remove a pending search if it exists
+    if (this.callSearch) this.callSearch.abort();
+
+    // initial render to show the loading spinner
+    this.render();
+
+    // query the search provider
+    this.callSearch = $.get({
+          url: searchURL,
+          dataType: 'json',
+        })
+        .done((data, status, xhr) => {
+        // make sure minimal data is present
```

Comment:
Is it possible for done to fire and have a statusText of abort? If the call was aborted, wouldn't it fail instead?